### PR TITLE
Add: Add sender from name newsletter setting

### PR DESCRIPTION
**Pull request by enejb**
Related to https://github.com/Automattic/jetpack/pull/37362 
## Proposed Changes

* This Pr ads from name to the calypso newsletter setting. 

Screenshot 2024-05-28 at 2 36 43 PM

## Why are these changes being made?

* Adds the new setting. 
* Adds the track event for that setting. 
* Adds form-input-validation hasIcon={ false } property which make is so that you can have form input validation also without an icon. 

## Testing Instructions

* Visit Settings/Newsletter and notice the new setting. 
* Change it make sure that the example preview updates as expected. 
* Save the value. 
* Make sure that the subscription email goes out with the new from name.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by pkuliga — COMMENTED** [review]
Tested it on localhost and Calypso Live and the sender name doesn't save properly for me. It stays there when I click the "Save settings" button but it's gone after I reload the page.

File: client/my-sites/site-settings/settings-newsletter/SenderNameSetting.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+export const SenderNameSetting = ( {
+	value = '',
+	disabled,
+	replyToValue,
+	updateFields,
+}: SenderNameSettingProps ) => {
+	const translate = useTranslate();
+	const selectedSite = useSelector( getSelectedSite );
+	const placeholder = selectedSite?.name || translate( 'Sender name' );
```
Comment (simison) [inline client/my-sites/site-settings/settings-newsletter/SenderNameSetting.tsx:36]: We could fall-back to the domain name here, and also in the backend? 🤔

Comment (enejb) [reply client/my-sites/site-settings/settings-newsletter/SenderNameSetting.tsx:36]: Good idea! Will double check this.

**Comment by enejb** [comment]
Thanks for testing it with simple sites @pkuliga. 
I thought it worked as expcted for me but I must have only been looking at jetpack/atomic sites. 

Here it the PR that fixes it on simple sites. https://github.com/Automattic/jetpack/pull/37586

**Review by simison — APPROVED** [review]

**Pull request merged**